### PR TITLE
Add `--log-level` command line option and deprecate `--debug`

### PR DESCRIPTION
**Pull request by saschagrunert**
The previous log-level implementation does not seem to work, which is
now fixed and aligns to other projects like podman and CRI-O.

Added documentation and integration testing as well.

### CODE REVIEW

**Comment by rhatdan** [comment]
I think if we are going to do this, we should remove (hide) `--debug` flag, since `--log-level debug` is the same thing, and becomes the default command users should use.

They should conflict, but removing --debug from man pages and hiding it from the CLI, is required to add --log-level

**Comment by vrothberg** [comment]
@rhatdan wins :^)

**Comment by TomSweeneyRedHat** [comment]
@saschagrunert when you said log-level didn't work, you just mean that the command wasn't there to do something like '--log-level=debug' right?  But the `--debug` flag is working as expected right?

Confession time, I hate the `--log-level` option, I much prefer the bare `--debug`.  But that being said, I'd be fine with this change as long as we keep --debug.  I'm prefer not hiding --debug, but am OK if it is hidden.  As long as my little crutch doesn't disappear!

**Comment by saschagrunert** [comment]
> @saschagrunert when you said log-level didn't work, you just mean that the command wasn't there to do something like '--log-level=debug' right? But the `--debug` flag is working as expected right?

Yes, something like `--log-level=...` does currently not work because the flag is not registered. `--debug` works as intended right now.

> Confession time, I hate the `--log-level` option, I much prefer the bare `--debug`. But that being said, I'd be fine with this change as long as we keep --debug. I'm prefer not hiding --debug, but am OK if it is hidden. As long as my little crutch doesn't disappear!

I mean we could also remove `--log-level` completely to only stick to `--debug`. But then we would not be able to show only warnings or info messages via logrus.

**Comment by rhatdan** [comment]
I think we should move to log-level to be consistent with Podman and to allow the flexibility that @saschagrunert just pointed out.

**Comment by saschagrunert** [comment]
Alright this is now a bit larger than expected, so let's see if the tests are passing.

File: docs/buildah-info.md [hunk 1]
```diff
@@ -10,8 +10,6 @@ buildah\-info - Display Buildah system information.
 The information displayed pertains to the host and current storage statistics which is useful when reporting issues.
 
 ## OPTIONS
-**--debug, -D**
-Show additional information.
 
 **--format** *template*
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply docs/buildah-info.md]: Ah you're right, it works. But is is not documented in the CLI output, which confused me a bit. I readded it.

File: tests/basic.bats [hunk 2]
```diff
@@ -102,9 +102,9 @@ load helpers
   run_buildah rmi remove-container-image
   run_buildah rmi containers-storage:other-new-image
   run_buildah rmi another-new-image
-  run_buildah --debug=false images -q
+  run_buildah --log-level=error images -q
   [ "$output" != "" ]
   run_buildah rmi -a
-  run_buildah --debug=false images -q
+  run_buildah --log-level=error images -q
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply tests/basic.bats]: It's not required, but I thought that it is now the _recommended way_ so the tests should reflect that too.

**Comment by saschagrunert** [comment]
> Just wanted to verify that the --debug option works as it did before, it's just not advertised. Is that correct @saschagruneret?

Yes, this is the intended behavior. :+1:

**Comment by saschagrunert** [comment]
Looks like a timeout issue in homu.

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot retry

**Pull request closed without merging**